### PR TITLE
Prepared statements for Classificationstore groups fix

### LINKED ISSUE #8865: Classification store field only shows one group since pimcore 6.8.12
# Bug Report

### Expected behavior
When an object has a classification store field with multiple active groups we expect that all those groups are shown in the object edit view.
image

### Actual behavior
Only the group with id = 1 is display in the object edit view (see screenshot)
image

### Steps to reproduce (Demo)
* Create classification store "test"
* Define classification store field on car class referencing the store "test"
* create multiple groups and attributes in the store "test"
* assign multiple groups to any car
* save and reload
* only one group is diplayed

I think the bug was introduced in 6.8.12 with the following line:
https://github.com/pimcore/pimcore/blob/v6.8.12/models/DataObject/ClassDefinition/Data/Classificationstore.php#L1231

This leads to a MySQL-Query as shown (note the single quotes around the list of IDs):
Selection_086

Back in 6.8.11 it wasn't a prepared statement but a sketchy string concatenation (https://github.com/pimcore/pimcore/blob/v6.8.10/models/DataObject/ClassDefinition/Data/Classificationstore.php#L1251). The MySQL-Query looked like this: 
Selection_087

### Possible solution
something like this:
```php
$groupList->setCondition('`id` in (' . implode(',', array_map(function ($id) { return '?'; }, $filteredGroupIds)) . ')', $filteredGroupIds);
```
=> produces a query with the same amount of bind variables like IDs in `$filteredGroupIds` and provides $filteredGroupIds as parameters

### PR DESCRIPTION
**Pull request by NiklasBr**
## Changes in this pull request  
Resolves #8865

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Classificationstore.php [hunk 1]
```diff
@@ -1238,7 +1238,8 @@ public function enrichLayoutDefinition($object, $context = [])
         $filteredGroupIds = array_keys($activeGroupIds, true, true);
 
         $groupList = new DataObject\Classificationstore\GroupConfig\Listing();
-        $groupList->setCondition('`id` in (?)', implode(',', $filteredGroupIds));
+        $groupList->setCondition('`id` in (' . implode(',', array_map(function ($id) { return '?'; }, $filteredGroupIds)) . ')', $filteredGroupIds);
```
Comment (jdreesen) [inline models/DataObject/ClassDefinition/Data/Classificationstore.php:1241]:
```suggestion
        $groupList->setCondition('`id` in (' . implode(',', array_fill(0, count($filteredGroupIds), '?')) . ')', $filteredGroupIds);
```

File: models/DataObject/ClassDefinition/Data/Classificationstore.php [hunk 2]
```diff
@@ -1304,7 +1305,7 @@ public function enrichLayoutDefinition($object, $context = [])
             $collectionIds = array_values($groupCollectionMapping);
 
             $relation = new DataObject\Classificationstore\CollectionGroupRelation\Listing();
-            $relation->setCondition('`colId` IN (?)', implode(',', $collectionIds));
+            $relation->setCondition('`colId` in (' . implode(',', array_map(function ($id) { return '?'; }, $collectionIds)) . ')', $collectionIds);
```
Comment (jdreesen) [inline models/DataObject/ClassDefinition/Data/Classificationstore.php:1308]:
```suggestion
            $relation->setCondition('`colId` in (' . implode(',', array_fill(0, count($collectionIds), '?')) . ')', $collectionIds);
```

**Review by jdreesen — CHANGES_REQUESTED** [review]

**Comment by NiklasBr** [comment]
Thanks @jdreesen!

**Comment by jdreesen** [comment]
Note that target branch and milestone currently differ.

**Comment by NiklasBr** [comment]
Yes, I targeted the wrong branch at first.

**Comment by jdreesen** [comment]
@weisswurstkanone you should probably adjust the milestone...

**Pull request merged**